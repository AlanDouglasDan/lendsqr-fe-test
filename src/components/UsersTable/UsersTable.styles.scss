.paginatedTable {
  background-color: #fff;
  padding: 30px;
  border: 1px solid rgba(33, 63, 125, 0.06);
  box-shadow: 3px 5px 20px rgba(0, 0, 0, 0.04);
  border-radius: 4px;
  overflow-x: auto;
  //   max-height: 640px;
  //   overflow-y: scroll;
}

table {
  white-space: nowrap;
}

.ellipsisIcon {
  color: #545f7d;
}

.py-22px {
  padding-top: 22px !important;
  padding-bottom: 22px !important;
}

.tableHeader {
  color: #545f7d;
  font-weight: 600;
  border: 0;
  text-transform: uppercase;
}

tbody {
  color: #545f7d;
}

.form-dropdown {
  width: 270px !important;
}

.cursor-pointer {
  cursor: pointer;
}

.light-yellow-btn {
  background-color: rgba(233, 178, 0, 0.1);
  border-radius: 100px;
  padding: 7px 18px;
  color: #e9b200;
}

.light-green-btn {
  background-color: rgba(57, 205, 98, 0.06);
  border-radius: 100px;
  padding: 7px 18px;
  color: #39cd62;
}

.light-grey-btn {
  background-color: rgba(84, 95, 125, 0.06);
  border-radius: 100px;
  padding: 7px 18px;
  color: #545f7d;
}

.light-red-btn {
  background-color: rgba(228, 3, 59, 0.1);
  border-radius: 100px;
  padding: 7px 18px;
  color: #e4033b;
}

.dropdown-toggle::after {
  display: none;
}

.dropdown-menu {
  border: 1px solid rgba(84, 95, 125, 0.04);
  box-shadow: 3px 5px 20px rgba(0, 0, 0, 0.04);
  border-radius: 4px;
  width: 180px;
  position: absolute;
}

.dropdown-item {
  cursor: pointer;
}

.form-dropdown .form-label {
  text-transform: capitalize;
}

.mt-20px {
  margin-top: 20px;
}

.no-underline {
  text-decoration: none;
}
