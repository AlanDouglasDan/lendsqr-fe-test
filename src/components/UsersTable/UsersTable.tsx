import { useState, useEffect } from "react";

// import useFetch from "../../hooks/useFetch";
import "./UsersTable.styles.scss";
import FilterForm from "components/FilterForm/FilterForm";
import ActionMenu from "components/ActionMenu/ActionMenu";
import Pagination from "components/Pagination/Pagination";
import { FilterIcon } from "components/icons";
import { OPTIONS, TABLE_HEADERS } from "../../constants";
import { UserInterface, FilterInterface } from "types/user.types";

const pending = <span className="light-yellow-btn">Pending</span>;
const active = <span className="light-green-btn">Active</span>;
const inactive = <span className="light-grey-btn">Inactive</span>;
const blacklisted = <span className="light-red-btn">Blacklisted</span>;

const initialState: FilterInterface = {
  organization: "",
  username: "",
  email: "",
  phoneNumber: "",
  date: "",
  status: "",
};

type Props = {
  data: UserInterface[];
  fetchUsers: () => void;
};

const UsersTable: React.FC<Props> = ({ data, fetchUsers }) => {
  const [rowsPerPage, setRowsPerPage] = useState<number>(10);
  const [page, setPage] = useState<number>(1);

  const [dataFiltered, setDataFiltered] = useState<UserInterface[]>([]);

  const [filters, setFilters] = useState<FilterInterface>(initialState);

  useEffect(() => {
    setDataFiltered(data);
  }, [data]);

  // populate the statuses array randomly with statuses of pending, active, inactive and blacklisted for each user
  const _statusesArray = data.map(() => {
    let status;
    const randomStatus = Math.floor(Math.random() * 4);
    if (randomStatus === 0) status = pending;
    else if (randomStatus === 1) status = active;
    else if (randomStatus === 2) status = inactive;
    else status = blacklisted;
    return status;
  });

  const [statusesArray, setStatusesArray] =
    useState<Array<JSX.Element>>(_statusesArray);

  // look through all the _data and get the organizations
  const _organizations: String[] = data?.map((item: any) => {
    return item.orgName;
  });

  // console.log(filters);
  // console.log(data);

  const numberOfPages = Math.ceil(data.length / rowsPerPage);

  const runFilterFunction = () => {
    const _data = data.filter((item: any) => {
      let match = true;

      if (filters.organization)
        if (item.orgName !== filters.organization) match = false;

      if (filters.username)
        if (item.username !== filters.username) match = false;

      if (filters.email) if (item.email !== filters.email) match = false;

      if (filters.phoneNumber)
        if (item.phoneNumber !== filters.phoneNumber) match = false;

      return match;
    });

    setDataFiltered(_data);
  };

  const resetFilterFunction = () => {
    setFilters(initialState);
    fetchUsers();
  };

  return (
    <>
      <div className="paginatedTable">
        <table className="table">
          <thead>
            <tr>
              {TABLE_HEADERS.map((header: any, index: any) => (
                <td className="tableHeader" key={index}>
                  <div className="d-flex">
                    <span>{header}</span>
                    <span className="dropdown">
                      <div
                        data-bs-toggle="dropdown"
                        className="dropdown-toggle"
                      >
                        <FilterIcon />
                      </div>
                      <FilterForm
                        filters={filters}
                        setFilters={setFilters}
                        _organizations={_organizations}
                        resetFilterFunction={resetFilterFunction}
                        runFilterFunction={runFilterFunction}
                      />
                    </span>
                  </div>
                </td>
              ))}
            </tr>
          </thead>
          <tbody>
            {dataFiltered
              ?.slice(
                (page - 1) * rowsPerPage,
                (page - 1) * rowsPerPage + rowsPerPage
              )
              .map((user: any, index: any) => {
                const date: Date = new Date(user.createdAt);

                // Convert the date to the desired format
                const formattedDate = date.toLocaleString("en-US", OPTIONS);

                return (
                  <tr key={user.id} className="align-middle">
                    <td
                      className={`py-22px ${
                        index === page * rowsPerPage - 1 && "border-0"
                      }`}
                    >
                      {user.orgName}
                    </td>
                    <td
                      className={`py-22px ${
                        index === page * rowsPerPage - 1 && "border-0"
                      }`}
                    >
                      {user.userName}
                    </td>
                    <td
                      className={`py-22px ${
                        index === page * rowsPerPage - 1 && "border-0"
                      }`}
                    >
                      {user.email}
                    </td>
                    <td
                      className={`py-22px ${
                        index === page * rowsPerPage - 1 && "border-0"
                      }`}
                    >
                      {user.phoneNumber}
                    </td>
                    <td
                      className={`py-22px ${
                        index === page * rowsPerPage - 1 && "border-0"
                      }`}
                    >
                      {formattedDate}
                    </td>
                    <td
                      className={`py-22px ${
                        index === page * rowsPerPage - 1 && "border-0"
                      }`}
                    >
                      {statusesArray[index]}
                    </td>
                    <td
                      className={`py-22px ${
                        index === page * rowsPerPage - 1 && "border-0"
                      }`}
                    >
                      <ActionMenu
                        userId={user.id}
                        status={statusesArray[index]}
                        setStatusesArray={setStatusesArray}
                        index={index}
                        statusesArray={statusesArray}
                        blacklisted={blacklisted}
                        active={active}
                        inactive={inactive}
                      />
                    </td>
                  </tr>
                );
              })}
          </tbody>
        </table>
      </div>

      <Pagination
        page={page}
        setPage={setPage}
        numberOfPages={numberOfPages}
        numRecords={String(data.length)}
        rowsPerPage={rowsPerPage}
        setRowsPerPage={setRowsPerPage}
      />
    </>
  );
};

export default UsersTable;
