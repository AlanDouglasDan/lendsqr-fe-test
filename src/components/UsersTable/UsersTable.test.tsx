import { BrowserRouter } from "react-router-dom";
import { render } from "@testing-library/react";

import UsersTable from "./UsersTable";

describe("Users Table Component Test", () => {
  it("Users Table Component snapshot", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <UsersTable data={[]} setData={() => {}} fetchUsers={() => {}} />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
