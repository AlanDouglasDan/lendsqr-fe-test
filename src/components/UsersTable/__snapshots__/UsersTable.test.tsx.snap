// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users Table Component Test Users Table Component snapshot 1`] = `
<DocumentFragment>
  <div
    class="paginatedTable"
  >
    <table
      class="table"
    >
      <thead>
        <tr>
          <td
            class="tableHeader"
          >
            <div
              class="d-flex"
            >
              <span>
                Organization
              </span>
              <span
                class="dropdown"
              >
                <div
                  class="dropdown-toggle"
                  data-bs-toggle="dropdown"
                >
                  <svg
                    class="ms-2 cursor-pointer"
                    fill="none"
                    height="12"
                    viewBox="0 0 16 12"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M6.22222 11.3333H9.77778V9.55554H6.22222V11.3333ZM0 0.666656V2.44443H16V0.666656H0ZM2.66667 6.88888H13.3333V5.1111H2.66667V6.88888Z"
                      fill="#545F7D"
                    />
                  </svg>
                </div>
                <div
                  class="dropdown-menu p-4 form-dropdown"
                >
                  <div
                    class="mb-3"
                  >
                    <label
                      class="form-label"
                      for="exampleDropdownFormOrganization"
                    >
                      Organization
                    </label>
                    <select
                      aria-label="Default select example"
                      class="form-select mb-3 form-label"
                      style="font-size: 1rem;"
                    >
                      <option>
                        Select
                      </option>
                    </select>
                    <label
                      class="form-label"
                      for="exampleDropdownFormUsername"
                    >
                      Username
                    </label>
                    <input
                      class="form-control"
                      id="exampleDropdownFormUsername"
                      placeholder="Username"
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    class="mb-3"
                  >
                    <label
                      class="form-label"
                      for="exampleDropdownFormEmail2"
                    >
                      Email
                    </label>
                    <input
                      class="form-control"
                      id="exampleDropdownFormEmail2"
                      placeholder="Email"
                      type="email"
                      value=""
                    />
                  </div>
                  <div
                    class="mb-3"
                  >
                    <label
                      class="form-label"
                      for="datepicker"
                    >
                      Date
                    </label>
                    <input
                      class="form-control"
                      id="datepicker"
                      type="date"
                      value=""
                    />
                  </div>
                  <div
                    class="mb-3"
                  >
                    <label
                      class="form-label"
                      for="exampleDropdownFormPhoneNumber"
                    >
                      Phone Number
                    </label>
                    <input
                      class="form-control"
                      id="exampleDropdownFormPhoneNumber"
                      placeholder="Phone Number"
                      type="text"
                      value=""
                    />
                  </div>
                  <label
                    class="form-label"
                    for="exampleDropdownFormStatus"
                  >
                    Status
                  </label>
                  <select
                    aria-label="Default select example"
                    class="form-select mb-3 form-label"
                    style="font-size: 1rem;"
                  >
                    <option>
                      Pending
                    </option>
                    <option
                      value="active"
                    >
                      Active
                    </option>
                    <option
                      value="inactive"
                    >
                      Inactive
                    </option>
                    <option
                      value="blacklisted"
                    >
                      Blacklisted
                    </option>
                  </select>
                  <div
                    class="grid-2 mt-4"
                  >
                    <button
                      class="btn btn-outline w-100"
                    >
                      Reset
                    </button>
                    <button
                      class="btn btn-themed w-100"
                    >
                      Filter
                    </button>
                  </div>
                </div>
              </span>
            </div>
          </td>
          <td
            class="tableHeader"
          >
            <div
              class="d-flex"
            >
              <span>
                Username
              </span>
              <span
                class="dropdown"
              >
                <div
                  class="dropdown-toggle"
                  data-bs-toggle="dropdown"
                >
                  <svg
                    class="ms-2 cursor-pointer"
                    fill="none"
                    height="12"
                    viewBox="0 0 16 12"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M6.22222 11.3333H9.77778V9.55554H6.22222V11.3333ZM0 0.666656V2.44443H16V0.666656H0ZM2.66667 6.88888H13.3333V5.1111H2.66667V6.88888Z"
                      fill="#545F7D"
                    />
                  </svg>
                </div>
                <div
                  class="dropdown-menu p-4 form-dropdown"
                >
                  <div
                    class="mb-3"
                  >
                    <label
                      class="form-label"
                      for="exampleDropdownFormOrganization"
                    >
                      Organization
                    </label>
                    <select
                      aria-label="Default select example"
                      class="form-select mb-3 form-label"
                      style="font-size: 1rem;"
                    >
                      <option>
                        Select
                      </option>
                    </select>
                    <label
                      class="form-label"
                      for="exampleDropdownFormUsername"
                    >
                      Username
                    </label>
                    <input
                      class="form-control"
                      id="exampleDropdownFormUsername"
                      placeholder="Username"
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    class="mb-3"
                  >
                    <label
                      class="form-label"
                      for="exampleDropdownFormEmail2"
                    >
                      Email
                    </label>
                    <input
                      class="form-control"
                      id="exampleDropdownFormEmail2"
                      placeholder="Email"
                      type="email"
                      value=""
                    />
                  </div>
                  <div
                    class="mb-3"
                  >
                    <label
                      class="form-label"
                      for="datepicker"
                    >
                      Date
                    </label>
                    <input
                      class="form-control"
                      id="datepicker"
                      type="date"
                      value=""
                    />
                  </div>
                  <div
                    class="mb-3"
                  >
                    <label
                      class="form-label"
                      for="exampleDropdownFormPhoneNumber"
                    >
                      Phone Number
                    </label>
                    <input
                      class="form-control"
                      id="exampleDropdownFormPhoneNumber"
                      placeholder="Phone Number"
                      type="text"
                      value=""
                    />
                  </div>
                  <label
                    class="form-label"
                    for="exampleDropdownFormStatus"
                  >
                    Status
                  </label>
                  <select
                    aria-label="Default select example"
                    class="form-select mb-3 form-label"
                    style="font-size: 1rem;"
                  >
                    <option>
                      Pending
                    </option>
                    <option
                      value="active"
                    >
                      Active
                    </option>
                    <option
                      value="inactive"
                    >
                      Inactive
                    </option>
                    <option
                      value="blacklisted"
                    >
                      Blacklisted
                    </option>
                  </select>
                  <div
                    class="grid-2 mt-4"
                  >
                    <button
                      class="btn btn-outline w-100"
                    >
                      Reset
                    </button>
                    <button
                      class="btn btn-themed w-100"
                    >
                      Filter
                    </button>
                  </div>
                </div>
              </span>
            </div>
          </td>
          <td
            class="tableHeader"
          >
            <div
              class="d-flex"
            >
              <span>
                Email
              </span>
              <span
                class="dropdown"
              >
                <div
                  class="dropdown-toggle"
                  data-bs-toggle="dropdown"
                >
                  <svg
                    class="ms-2 cursor-pointer"
                    fill="none"
                    height="12"
                    viewBox="0 0 16 12"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M6.22222 11.3333H9.77778V9.55554H6.22222V11.3333ZM0 0.666656V2.44443H16V0.666656H0ZM2.66667 6.88888H13.3333V5.1111H2.66667V6.88888Z"
                      fill="#545F7D"
                    />
                  </svg>
                </div>
                <div
                  class="dropdown-menu p-4 form-dropdown"
                >
                  <div
                    class="mb-3"
                  >
                    <label
                      class="form-label"
                      for="exampleDropdownFormOrganization"
                    >
                      Organization
                    </label>
                    <select
                      aria-label="Default select example"
                      class="form-select mb-3 form-label"
                      style="font-size: 1rem;"
                    >
                      <option>
                        Select
                      </option>
                    </select>
                    <label
                      class="form-label"
                      for="exampleDropdownFormUsername"
                    >
                      Username
                    </label>
                    <input
                      class="form-control"
                      id="exampleDropdownFormUsername"
                      placeholder="Username"
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    class="mb-3"
                  >
                    <label
                      class="form-label"
                      for="exampleDropdownFormEmail2"
                    >
                      Email
                    </label>
                    <input
                      class="form-control"
                      id="exampleDropdownFormEmail2"
                      placeholder="Email"
                      type="email"
                      value=""
                    />
                  </div>
                  <div
                    class="mb-3"
                  >
                    <label
                      class="form-label"
                      for="datepicker"
                    >
                      Date
                    </label>
                    <input
                      class="form-control"
                      id="datepicker"
                      type="date"
                      value=""
                    />
                  </div>
                  <div
                    class="mb-3"
                  >
                    <label
                      class="form-label"
                      for="exampleDropdownFormPhoneNumber"
                    >
                      Phone Number
                    </label>
                    <input
                      class="form-control"
                      id="exampleDropdownFormPhoneNumber"
                      placeholder="Phone Number"
                      type="text"
                      value=""
                    />
                  </div>
                  <label
                    class="form-label"
                    for="exampleDropdownFormStatus"
                  >
                    Status
                  </label>
                  <select
                    aria-label="Default select example"
                    class="form-select mb-3 form-label"
                    style="font-size: 1rem;"
                  >
                    <option>
                      Pending
                    </option>
                    <option
                      value="active"
                    >
                      Active
                    </option>
                    <option
                      value="inactive"
                    >
                      Inactive
                    </option>
                    <option
                      value="blacklisted"
                    >
                      Blacklisted
                    </option>
                  </select>
                  <div
                    class="grid-2 mt-4"
                  >
                    <button
                      class="btn btn-outline w-100"
                    >
                      Reset
                    </button>
                    <button
                      class="btn btn-themed w-100"
                    >
                      Filter
                    </button>
                  </div>
                </div>
              </span>
            </div>
          </td>
          <td
            class="tableHeader"
          >
            <div
              class="d-flex"
            >
              <span>
                Phone Number
              </span>
              <span
                class="dropdown"
              >
                <div
                  class="dropdown-toggle"
                  data-bs-toggle="dropdown"
                >
                  <svg
                    class="ms-2 cursor-pointer"
                    fill="none"
                    height="12"
                    viewBox="0 0 16 12"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M6.22222 11.3333H9.77778V9.55554H6.22222V11.3333ZM0 0.666656V2.44443H16V0.666656H0ZM2.66667 6.88888H13.3333V5.1111H2.66667V6.88888Z"
                      fill="#545F7D"
                    />
                  </svg>
                </div>
                <div
                  class="dropdown-menu p-4 form-dropdown"
                >
                  <div
                    class="mb-3"
                  >
                    <label
                      class="form-label"
                      for="exampleDropdownFormOrganization"
                    >
                      Organization
                    </label>
                    <select
                      aria-label="Default select example"
                      class="form-select mb-3 form-label"
                      style="font-size: 1rem;"
                    >
                      <option>
                        Select
                      </option>
                    </select>
                    <label
                      class="form-label"
                      for="exampleDropdownFormUsername"
                    >
                      Username
                    </label>
                    <input
                      class="form-control"
                      id="exampleDropdownFormUsername"
                      placeholder="Username"
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    class="mb-3"
                  >
                    <label
                      class="form-label"
                      for="exampleDropdownFormEmail2"
                    >
                      Email
                    </label>
                    <input
                      class="form-control"
                      id="exampleDropdownFormEmail2"
                      placeholder="Email"
                      type="email"
                      value=""
                    />
                  </div>
                  <div
                    class="mb-3"
                  >
                    <label
                      class="form-label"
                      for="datepicker"
                    >
                      Date
                    </label>
                    <input
                      class="form-control"
                      id="datepicker"
                      type="date"
                      value=""
                    />
                  </div>
                  <div
                    class="mb-3"
                  >
                    <label
                      class="form-label"
                      for="exampleDropdownFormPhoneNumber"
                    >
                      Phone Number
                    </label>
                    <input
                      class="form-control"
                      id="exampleDropdownFormPhoneNumber"
                      placeholder="Phone Number"
                      type="text"
                      value=""
                    />
                  </div>
                  <label
                    class="form-label"
                    for="exampleDropdownFormStatus"
                  >
                    Status
                  </label>
                  <select
                    aria-label="Default select example"
                    class="form-select mb-3 form-label"
                    style="font-size: 1rem;"
                  >
                    <option>
                      Pending
                    </option>
                    <option
                      value="active"
                    >
                      Active
                    </option>
                    <option
                      value="inactive"
                    >
                      Inactive
                    </option>
                    <option
                      value="blacklisted"
                    >
                      Blacklisted
                    </option>
                  </select>
                  <div
                    class="grid-2 mt-4"
                  >
                    <button
                      class="btn btn-outline w-100"
                    >
                      Reset
                    </button>
                    <button
                      class="btn btn-themed w-100"
                    >
                      Filter
                    </button>
                  </div>
                </div>
              </span>
            </div>
          </td>
          <td
            class="tableHeader"
          >
            <div
              class="d-flex"
            >
              <span>
                Date Joined
              </span>
              <span
                class="dropdown"
              >
                <div
                  class="dropdown-toggle"
                  data-bs-toggle="dropdown"
                >
                  <svg
                    class="ms-2 cursor-pointer"
                    fill="none"
                    height="12"
                    viewBox="0 0 16 12"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M6.22222 11.3333H9.77778V9.55554H6.22222V11.3333ZM0 0.666656V2.44443H16V0.666656H0ZM2.66667 6.88888H13.3333V5.1111H2.66667V6.88888Z"
                      fill="#545F7D"
                    />
                  </svg>
                </div>
                <div
                  class="dropdown-menu p-4 form-dropdown"
                >
                  <div
                    class="mb-3"
                  >
                    <label
                      class="form-label"
                      for="exampleDropdownFormOrganization"
                    >
                      Organization
                    </label>
                    <select
                      aria-label="Default select example"
                      class="form-select mb-3 form-label"
                      style="font-size: 1rem;"
                    >
                      <option>
                        Select
                      </option>
                    </select>
                    <label
                      class="form-label"
                      for="exampleDropdownFormUsername"
                    >
                      Username
                    </label>
                    <input
                      class="form-control"
                      id="exampleDropdownFormUsername"
                      placeholder="Username"
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    class="mb-3"
                  >
                    <label
                      class="form-label"
                      for="exampleDropdownFormEmail2"
                    >
                      Email
                    </label>
                    <input
                      class="form-control"
                      id="exampleDropdownFormEmail2"
                      placeholder="Email"
                      type="email"
                      value=""
                    />
                  </div>
                  <div
                    class="mb-3"
                  >
                    <label
                      class="form-label"
                      for="datepicker"
                    >
                      Date
                    </label>
                    <input
                      class="form-control"
                      id="datepicker"
                      type="date"
                      value=""
                    />
                  </div>
                  <div
                    class="mb-3"
                  >
                    <label
                      class="form-label"
                      for="exampleDropdownFormPhoneNumber"
                    >
                      Phone Number
                    </label>
                    <input
                      class="form-control"
                      id="exampleDropdownFormPhoneNumber"
                      placeholder="Phone Number"
                      type="text"
                      value=""
                    />
                  </div>
                  <label
                    class="form-label"
                    for="exampleDropdownFormStatus"
                  >
                    Status
                  </label>
                  <select
                    aria-label="Default select example"
                    class="form-select mb-3 form-label"
                    style="font-size: 1rem;"
                  >
                    <option>
                      Pending
                    </option>
                    <option
                      value="active"
                    >
                      Active
                    </option>
                    <option
                      value="inactive"
                    >
                      Inactive
                    </option>
                    <option
                      value="blacklisted"
                    >
                      Blacklisted
                    </option>
                  </select>
                  <div
                    class="grid-2 mt-4"
                  >
                    <button
                      class="btn btn-outline w-100"
                    >
                      Reset
                    </button>
                    <button
                      class="btn btn-themed w-100"
                    >
                      Filter
                    </button>
                  </div>
                </div>
              </span>
            </div>
          </td>
          <td
            class="tableHeader"
          >
            <div
              class="d-flex"
            >
              <span>
                Status
              </span>
              <span
                class="dropdown"
              >
                <div
                  class="dropdown-toggle"
                  data-bs-toggle="dropdown"
                >
                  <svg
                    class="ms-2 cursor-pointer"
                    fill="none"
                    height="12"
                    viewBox="0 0 16 12"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M6.22222 11.3333H9.77778V9.55554H6.22222V11.3333ZM0 0.666656V2.44443H16V0.666656H0ZM2.66667 6.88888H13.3333V5.1111H2.66667V6.88888Z"
                      fill="#545F7D"
                    />
                  </svg>
                </div>
                <div
                  class="dropdown-menu p-4 form-dropdown"
                >
                  <div
                    class="mb-3"
                  >
                    <label
                      class="form-label"
                      for="exampleDropdownFormOrganization"
                    >
                      Organization
                    </label>
                    <select
                      aria-label="Default select example"
                      class="form-select mb-3 form-label"
                      style="font-size: 1rem;"
                    >
                      <option>
                        Select
                      </option>
                    </select>
                    <label
                      class="form-label"
                      for="exampleDropdownFormUsername"
                    >
                      Username
                    </label>
                    <input
                      class="form-control"
                      id="exampleDropdownFormUsername"
                      placeholder="Username"
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    class="mb-3"
                  >
                    <label
                      class="form-label"
                      for="exampleDropdownFormEmail2"
                    >
                      Email
                    </label>
                    <input
                      class="form-control"
                      id="exampleDropdownFormEmail2"
                      placeholder="Email"
                      type="email"
                      value=""
                    />
                  </div>
                  <div
                    class="mb-3"
                  >
                    <label
                      class="form-label"
                      for="datepicker"
                    >
                      Date
                    </label>
                    <input
                      class="form-control"
                      id="datepicker"
                      type="date"
                      value=""
                    />
                  </div>
                  <div
                    class="mb-3"
                  >
                    <label
                      class="form-label"
                      for="exampleDropdownFormPhoneNumber"
                    >
                      Phone Number
                    </label>
                    <input
                      class="form-control"
                      id="exampleDropdownFormPhoneNumber"
                      placeholder="Phone Number"
                      type="text"
                      value=""
                    />
                  </div>
                  <label
                    class="form-label"
                    for="exampleDropdownFormStatus"
                  >
                    Status
                  </label>
                  <select
                    aria-label="Default select example"
                    class="form-select mb-3 form-label"
                    style="font-size: 1rem;"
                  >
                    <option>
                      Pending
                    </option>
                    <option
                      value="active"
                    >
                      Active
                    </option>
                    <option
                      value="inactive"
                    >
                      Inactive
                    </option>
                    <option
                      value="blacklisted"
                    >
                      Blacklisted
                    </option>
                  </select>
                  <div
                    class="grid-2 mt-4"
                  >
                    <button
                      class="btn btn-outline w-100"
                    >
                      Reset
                    </button>
                    <button
                      class="btn btn-themed w-100"
                    >
                      Filter
                    </button>
                  </div>
                </div>
              </span>
            </div>
          </td>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
  <div
    class="d-flex align-items-center justify-content-between mt-20px center-grid"
  >
    <div
      class="d-flex align-items-center"
    >
      <div>
        Showing
      </div>
      <div
        class="dropdown"
      >
        <button
          aria-expanded="false"
          class="dropdown-toggle mx-2 rowsPerPageBtn"
          data-bs-toggle="dropdown"
          type="button"
        >
          10
          <i
            class="fa fa-angle-down ms-2"
          />
        </button>
        <ul
          class="dropdown-menu"
        >
          <li
            class="dropdown-item"
          >
            5
          </li>
          <li
            class="dropdown-item"
          >
            10
          </li>
          <li
            class="dropdown-item"
          >
            20
          </li>
        </ul>
      </div>
      <div>
        out of 0
      </div>
    </div>
    <div
      class="d-flex align-items-center"
    >
      <button
        class="operandButton me-3"
        disabled=""
      >
        <i
          class="fa fa-angle-left"
        />
      </button>
      <div
        class="paginationNavigator activePaginationNavigator"
      >
        1
      </div>
      <div>
        ...
      </div>
      <div
        class="paginationNavigator false"
      >
        0
      </div>
      <div
        class="paginationNavigator activePaginationNavigator"
      >
        1
      </div>
      <div
        class="paginationNavigator false"
      >
        2
      </div>
      <div>
        ...
      </div>
      <div
        class="paginationNavigator false"
      >
        0
      </div>
      <button
        class="operandButton ms-3"
      >
        <i
          class="fa fa-angle-right"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
